cmd/clusterctl/clusterdeployer/bootstrap/kind/kind_test.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kind
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func TestCreate(t *testing.T) {
+	var testcases = []struct {
+		name      string
+		execError error
+		expectErr bool
+	}{
+		{
+			name: "success",
+		},
+		{
+			name:      "exec fail",
+			execError: fmt.Errorf("test error"),
+			expectErr: true,
+		},
+	}
+	for _, testcase := range testcases {
+		t.Run(testcase.name, func(t *testing.T) {
+			m := New()
+			m.execFunc = func(args ...string) (string, error) {
```

Comment:
code in k/k uses a slightly different pattern for the sake of unit testing.
it would declare a Kind interface. the core logic defines an object with the actual methods and the unit tests define one with fake methods.

haven't looked at what the rest of cluster-api is doing.

Reply:
I got this from the minikube bootstrapper, which uses this approach. I'm ok changing it if needed.

Reply:
my vote would be to use the "mocking / interface" pattern instead, but would wait for others to comment too.